test/bigfloat_dot.jl

```diff
@@ -0,0 +1,81 @@
+# Copyright (c) 2023 MutableArithmetics.jl contributors
+#
+# This Source Code Form is subject to the terms of the Mozilla Public License,
+# v.2.0. If a copy of the MPL was not distributed with this file, You can obtain
+# one at http://mozilla.org/MPL/2.0/.
+
+backup_bigfloats(v::AbstractVector{BigFloat}) = map(x -> x + 0, v)
+
+absolute_error(accurate::Real, approximate::Real) = abs(accurate - approximate)
+
+function relative_error(accurate::Real, approximate::Real)
+    return absolute_error(accurate, approximate) / abs(accurate)
+end
+
+function dotter(x::V, y::V) where {V<:AbstractVector{<:Real}}
+    let x = x, y = y
+        () -> LinearAlgebra.dot(x, y)
+    end
+end
+
+function reference_dot(x::V, y::V) where {F<:Real,V<:AbstractVector{F}}
+    return setprecision(dotter(x, y), F, 8 * precision(F))
+end
+
+function dot_test_relative_error(x::V, y::V) where {V<:AbstractVector{BigFloat}}
+    buf = MA.buffer_for(LinearAlgebra.dot, V, V)
+
+    input = (x, y)
+    backup = map(backup_bigfloats, input)
+
+    output = BigFloat()
+
+    MA.buffered_operate_to!!(buf, output, LinearAlgebra.dot, input...)
```

Comment:
Does this allocate ? If not, you can use what is in `test/utils.Jl` to test that it does not allocate